pkg/operator/controller/ingress_status_test.go

```diff
@@ -0,0 +1,262 @@
+package controller
+
+import (
+	"fmt"
+	"testing"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestComputeIngressStatusConditions(t *testing.T) {
+	testCases := []struct {
+		description     string
+		availRepl, repl int32
```

Comment:
You could encode the expected condition into this as well, so your cases would be like

```go
{"0/2 deployment replicas available", 0, 2, OperatorStatusTypeAvailable, ConditionFalse},
{"1/2 deployment replicas available", 1, 2, OperatorStatusTypeAvailable, ConditionTrue},
{"2/2 deployment replicas available", 2, 2, OperatorStatusTypeAvailable, ConditionTrue},
```

And, later, could be expanded to encompass asserting a whole set of conditions.

Perhaps more importantly, another advantage is that you avoid explicitly re-implementing the condition algorithm inside the test code (e.g. as written, L35 would have to eternally mirror the internals of `computeIngressStatusConditions`)

Reply:
Thanks for the input. I have updated the test cases as you requested.